src/components/flow/routers/webhook/WebhookRouterForm.tsx

```diff
@@ -208,10 +208,15 @@ export default class WebhookRouterForm extends React.Component<
 
   private handleSave(): void {
     // validate our url in case they haven't interacted
-    const valid = this.handleUpdate(
-      { url: this.state.url.value, resultName: this.state.resultName.value },
-      true
-    );
+    let valid = false;
+    if (this.state.method.value.name === 'FUNCTION') {
+      valid = this.handleUpdate({ resultName: this.state.resultName.value }, true);
+    } else {
+      valid = this.handleUpdate(
+        { url: this.state.url.value, resultName: this.state.resultName.value },
+        true
+      );
```

Comment:
Why we need this condition?

Reply:
@pankaj-ag If the method is of type function then we only check that the result name should be required. Otherwise, we check that the URL and result name both should be required.

Reply:
Okay, I think we can simplify this thing by making the URL required as a text box, but let's do that later.